platform/stellar/api.go

```diff
@@ -40,34 +42,61 @@ func (p *Platform) CurrentBlockNumber() (int64, error) {
 
 func (p *Platform) GetBlockByNumber(num int64) (*blockatlas.Block, error) {
 	if srcBlock, err := p.client.GetBlockByNumber(num); err == nil {
-		block := NormalizeBlock(srcBlock, p.CoinIndex)
+		block := NormalizeBlock(srcBlock, *p)
 		return &block, nil
 	} else {
 		return nil, err
 	}
 }
 
-func NormalizeBlock(block *Block, nativeCoinIndex uint) blockatlas.Block {
+func NormalizeBlock(block *Block, p Platform) blockatlas.Block {
 	return blockatlas.Block{
 		ID:     block.Ledger.Id,
 		Number: block.Ledger.Sequence,
-		Txs:    NormalizePayments(block.Payments, nativeCoinIndex),
+		Txs:    NormalizePayments(block.Payments, p),
 	}
 }
 
-func NormalizePayments(payments []Payment, nativeCoinIndex uint) (txs []blockatlas.Tx) {
+func NormalizePayments(payments []Payment, p Platform) (txs []blockatlas.Tx) {
+	var wg sync.WaitGroup
+	tupleChan := make(chan struct {
+		Payment;
+		TrxHash
+	})
+
 	for _, payment := range payments {
-		tx, ok := Normalize(&payment, nativeCoinIndex)
-		if !ok {
-			continue
-		}
-		txs = append(txs, tx)
+		wg.Add(1)
+		go func(id string) {
+			defer wg.Done()
+			hash, err := p.client.GetTrxHash(id)
+			if err != nil {
+				return
+			}
+			tupleChan <- struct {
+				Payment;
+				TrxHash
+			}{payment, hash}
```

Comment:
`payment` here this is not thread-safe. Try to pass all object to the method instead of the hash

Reply:
https://github.com/trustwallet/blockatlas/pull/600/commits/ce18f9ca43b4ed373e490e968d47e20440d97ff0 also changed here https://github.com/trustwallet/blockatlas/pull/602